plugins/outputs/signalfx/signalfx.go

```diff
@@ -95,10 +96,10 @@ func (s *SignalFx) Connect() error {
 
 /*Close closes the connection to SignalFx*/
 func (s *SignalFx) Close() error {
-	s.ctx.Done()
-	s.client = nil
-	close(s.done)
-	s.wg.Wait()
+	close(s.done)  // drain the input channels
+	s.wg.Wait()    // wait for the input channels to be drained
+	s.ctx.Done()   // safely close the sink context
```

Comment:
`ctx.Done()` returns a channel that will be closed when the context is terminated.  It doesn't do anything in and of itself.

Reply:
Also I don't understand the point of having a context and a done channel.  You should probably just use a cancellable context.